zerver/views/upload.py

```diff
@@ -82,10 +82,12 @@ def serve_file(request: HttpRequest, user_profile: UserProfile,
 
     return serve_s3(request, path_id, url_only)
 
-def serve_local_file_unauthed(request: HttpRequest, token: str) -> HttpResponse:
+def serve_local_file_unauthed(request: HttpRequest, token: str, filename: str) -> HttpResponse:
     path_id = get_local_file_path_id_from_token(token)
     if path_id is None:
         return json_error(_("Invalid token"))
+    if path_id.split('/')[-1] != filename:
+        return json_error(_("Filename passed to the API endpoint doesn't match the uploaded file"))
```

Comment:
Maybe "Invalid filename"?  I think it's probably not worth asking our translators to translate this complex of a string.

Reply:
Makes sense, changed